docs/DeveloperGuide.adoc

```diff
@@ -312,16 +317,58 @@ The diagram below shows how the schedule command is parsed in the Logic componen
 .Logic diagram for `schedule EVENT` command
 image::ScheduleEventSdForLogic.png[width="800"]
 
+===== Cancel
+
+The `cancel` command is essentially the reverse of the schedule command, allowing the user to delete events from the calendar. It is implemented as such:
+
+* `cancel event/EVENT_NAME date/DATE start/START_TIME end/END_TIME`
+
+An example usage would be
+
+`cancel event/CS2103 Tutorial 1 date/23-01-2018 start/15:00 end/16:00`
+
+From there, the event with the specified fields will be located in the event list, and deleted accordingly.
+
+If the event is not present within the calendar, the user will receive an error message.
+
+Also, the interactions between the components work in much the same way as the schedule command.
+
+===== Undo/Redo
+
+The `undo` and `redo` command was extended to include the actions made by the user while interacting with the calendar.
+
+The undo/redo feature is also facilitated by a VersionedCalendar, and further details are as described in the undo/redo feature.
+
+The most pertinent aspect of this feature is how the model component handles the undo/redo. The ModelManager simply keeps track of _which_
+model -- address book or calendar -- had committed a change, and handles it accordingly whenever the user wishes to undo or redo.
```

Comment:
perhaps mention what data structure/method is used to track the combined history

Reply:
Added a section detailing this.